packages/catalog-process/src/services/catalogService.ts

```diff
@@ -696,24 +699,43 @@ export async function updateDocumentLogic({
   });
 }
 
-export function createDescriptorLogic({
+export async function createDescriptorLogic({
   eServiceId,
   eserviceDescriptorSeed,
   authData,
   eService,
+  getAttributeById,
 }: {
   eServiceId: string;
   eserviceDescriptorSeed: EServiceDescriptorSeed;
   authData: AuthData;
   eService: WithMetadata<EService> | undefined;
-}): CreateEvent<EServiceEvent> {
+  getAttributeById: (
+    id: string
+  ) => Promise<WithMetadata<Attribute> | undefined>;
+}): Promise<CreateEvent<EServiceEvent>> {
   assertEServiceExist(eServiceId, eService);
   assertRequesterAllowed(eService.data.producerId, authData.organizationId);
   hasNotDraftDescriptor(eService.data);
 
   const newVersion = nextDescriptorVersion(eService.data);
 
   const certifiedAttributes = eserviceDescriptorSeed.attributes.certified;
+  const declaredAttributes = eserviceDescriptorSeed.attributes.declared;
+  const verifiedAttributes = eserviceDescriptorSeed.attributes.verified;
+
+  const attributesSeeds = [
+    ...certifiedAttributes.flat(),
+    ...declaredAttributes.flat(),
+    ...verifiedAttributes.flat(),
+  ];
+
+  for (const attributeSeed of attributesSeeds) {
+    const attribute = await getAttributeById(attributeSeed.id);
+    if (attribute === undefined) {
+      throw attributeNotFound(attributeSeed.id);
+    }
+  }
```

Comment:
Querying each attribute one by one  could be very costly in terms of performance.
A better approach could be making just one query:
```typescript
const attributes = await readModel.attributes.find({ 'data.id': { $in: attributesIds } }).toArray()
```

I tested the performance of the two approaches locally, here the results:

<img width="319" alt="Screenshot 2024-02-06 alle 10 18 37" src="https://github.com/pagopa/interop-be-monorepo/assets/62668966/cb2b38a6-9d67-4ad0-bdf7-3812783cf7a7">

Reply:
I know, but there are a couple of considerations:
- is 100 a realistic number for this use case?
- by using your query, which error should I return? How to distinguish which attribute has not been found? Maybe we could check if `attributes.length < attributeIds.length`, but we might need some more precise info.

Reply:
You are right, I repeated the test with a more realistic value:

<img width="319" alt="Screenshot 2024-02-06 alle 15 00 55" src="https://github.com/pagopa/interop-be-monorepo/assets/62668966/82621f66-0185-45cf-8ba2-f409ed2fecc0">

The logic could be adapted as following:
```typescript
const attributes = await readModel.attributes.find({ 'data.id': { $in: attributesSeedIds } }).toArray()
const attributesIds = attributes.map(({ data }) => data.id)

for (const attributeSeedId of attributesSeedIds) {
  // If the attributeSeedId is not inside the array of query results throw the error
  if (!attributesIds.includes(attributeSeedId)) {
    throw attributeNotFound(attributeSeedId);
  }
}
```

Reply:
Updated